analyzers/src/SonarAnalyzer.CSharp/Helpers/PatternExpressionObjectToPatternMapping.cs

```diff
@@ -28,26 +28,25 @@ namespace SonarAnalyzer.Helpers
 {
     internal static class PatternExpressionObjectToPatternMapping
     {
+        /// <summary>
+        /// Maps the tuple arguments of <paramref name="expression"/> to the positional sub-pattern of <paramref name="pattern"/>.
+        /// For a pattern like: <code>(x, y) is (1, 2)</code>x is mapped to numeric literal 1 and y is mapped to 2.
+        /// </summary>
+        /// <param name="expression">A tuple expression.</param>
+        /// <param name="pattern">A pattern that can be matched to the tuple <paramref name="expression"/>.</param>
+        /// <param name="objectToPatternMap">The mapping between the tuple arguments and the positional sub-patterns.</param>
```

Comment:
This PR is quite big already and I changed `MapObjectToPattern` without even requiring it. If I change the signature now, the PR will become even bigger because 3 more call sites unrelated to the PR have to be modified. Can we do the changes to this method in a later separate PR?

Reply:
Sure